devito/passes/iet/misc.py

```diff
@@ -204,3 +207,39 @@ def remove_redundant_moddims(iet):
     iet = Transformer(subs, nested=True).visit(iet)
 
     return iet
+
+
+def abridge_dim_names(iet):
+    subs = {}
+    mapper = {}
+    tree = retrieve_iteration_tree(iet)
+    for it in tree:
+        indexeds = FindSymbols('indexeds').visit(it.inner)
+        inds = set().union(*[pull_dims(i, flag=False) for i in indexeds])
+        inds = [i for i in inds if any([d.is_Sub for d in i._defines])]
+        inds = [i for i in inds if not i.is_SubIterator]
```

Comment:
when we're sure they're dimensions, we typically use the letter `d` for iteration

I admit this is loosely done, though